tests/conftest.py

```diff
@@ -781,3 +781,14 @@ def navigation_key_expected_key_pair(request):
     The expected key is the one which is passed to the super `keypress` calls.
     """
     return request.param
+
+
+@pytest.fixture
+def widget_size(request):
+    """
+    Returns widget size for any widget.
+    """
+    widget = request.getfixturevalue(request.param)
+    widget_type, *_ = widget.sizing()
+    return (20, ) if widget_type == 'flow' else \
+           (200, 20) if widget_type == 'box' else None
```

Comment:
I've not seen the `indirect` approach; it looks a little unclear to someone who hasn't seen it before, though I can guess what is happening. Could this be translated to returning a function to be applied to a widget? That might be simpler? Potentially we could try caching the values, if that had a speed factor, though that might reduce test isolation.

Reply:
On second thought, this does look a bit unclear. I'll see if this can be translated into a simpler form.

Reply:
@neiljp I wonder if using a function would be less complicated?

```python
def widget_size(widget):
    """
    Returns widget size for any widget.
    """
    sizes = {
        'box': (200, 20),
        'flow': (20, ),
    }
    widget_type, *_ = widget.sizing()
    return sizes.get(widget_type, None)
```

In tests,  we could do:

```python
view.keypress(widget_size(widget), key)
```

However, this would require an import (unlike fixtures).

Reply:
We could resolve this by simply using a factory fixture:
https://docs.pytest.org/en/stable/fixture.html#factories-as-fixtures